src/core/execute.c

```diff
@@ -2925,6 +2942,9 @@ void exec_context_init(ExecContext *c) {
         c->umask = 0022;
         c->ioprio = IOPRIO_PRIO_VALUE(IOPRIO_CLASS_BE, 0);
         c->cpu_sched_policy = SCHED_OTHER;
+        c->std_input_fdname = strdup("stdin");
+        c->std_output_fdname = strdup("stdout");
+        c->std_error_fdname = strdup("stderr");
```

Comment:
Unfortunately every `strdup` needs to be checked. So this needs to be reworked to not allocate anything (that's a much better option than checking if the allocation succeeded, and then changing this function to return status, and propagating that down the stack). Let's instead use the convention that if `std_{input,output,error}_fdname` is NULL, "stdin", "stdout", and "stderr" should be used.